rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/model/security/UserPrincipal.java

```diff
@@ -0,0 +1,50 @@
+package org.niewidoczniakademicy.rezerwacje.model.security;
+
+import lombok.AllArgsConstructor;
+import org.niewidoczniakademicy.rezerwacje.model.database.SystemUser;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.List;
+
+@AllArgsConstructor
+public final class UserPrincipal implements UserDetails {
+    private final SystemUser user;
```

Comment:
Do we need this class or will we need it later? It seems we can just return an instance of org.springframework.security.core.userdetails.User class in UserService.loadUserByUsername method